<launch>

  <!--
  	This pushes the local PYTHONPATH into the launch file, so that the NaoQI API is found.
    You need to add the Nao's API dir to your PYTHONPATH so that the modules are found.
  -->
  <env name="PYTHONPATH" value="$(env PYTHONPATH)" /> 

  
  <!-- 
    Start Nao 1
  -->
  <arg name="nao_ip1" default="$(optenv NAO_IP 127.0.0.1)" />
  <arg name="nao_port1" default="$(optenv NAO_PORT 9560)" />
  
  <node pkg="nao_driver" type="nao_walker.py" name="nao_walker1" required="true" args="--pip=$(arg nao_ip1) --pport=$(arg nao_port1)" output="screen">
  	<remap from="cmd_vel" to="cmd_vel1"/>
  </node>
  
  <node pkg="nao_driver" type="nao_sensors.py" name="nao_sensors1" required="true" args="--pip=$(arg nao_ip1) --pport=$(arg nao_port1)" output="screen">
  	<remap from="torso_odometry" to="torso_odometry1"/>
  </node>
  
  <node pkg="nao_driver" type="nao_controller.py" name="nao_controller1" required="true" args="--pip=$(arg nao_ip1) --pport=$(arg nao_port1)" output="screen"/>

 <!--
    Top camera 1
  -->
  <node pkg="nao_driver" type="nao_camera.py" name="nao_camera1" required="true" args="--pip=$(arg nao_ip1) --pport=$(arg nao_port1)" output="screen">
    <param name="camera_switch" value="0"/>
    <param name="resolution" value="2"/>
    <remap from="image_raw" to="image_raw1"/>
  </node>
  
  <!--
    Sonar 1
  -->
  <node pkg="nao_custom" type="nao_sonar" name="nao_sonar1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip1)"/>
    <param name="NAO_PORT" value="$(arg nao_port1)"/>
    <remap from="sonar" to="sonar1"/>
  </node>
  
  <!--
    Nao detection 1
  -->
  <node pkg="collaborative_motion" type="nao_detection" name="nao_d1" args="1" output="screen">
    <param name="H_MIN" value="115"/>
    <param name="H_MAX" value="180"/>
    <param name="S_MIN" value="100"/>
    <param name="S_MAX" value="255"/>
    <param name="V_MIN" value="90"/>
    <param name="V_MAX" value="255"/>
  </node>
  
  <node pkg="collaborative_motion" type="nao_tracker_camera" name="nao_t_c1" args="1" output="screen"/>

  <!--
    Nao behavior 1 (hello)
  -->
  <node pkg="collaborative_motion" type="nao_hello" name="nao_hello1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip1)"/>
    <param name="NAO_PORT" value="$(arg nao_port1)"/>
  </node>
  
  <!--
    Nao controller 1
  -->
  <node pkg="collaborative_motion" type="nao_controller" name="nao_c1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip1)"/>
    <param name="NAO_PORT" value="$(arg nao_port1)"/>
  </node>
  
</launch>
